sw/device/tests/edn_boot_mode.c

```diff
@@ -0,0 +1,170 @@
+    // Enable EDN0 in auto request mode.
+    CHECK_DIF_OK(dif_edn_set_auto_mode(&edn0, edn_params0));
+  }
+
+  if (round == 2) {
+    // Enable EDN1 (the one connected to OTBN RND) in auto request mode.
+    CHECK_DIF_OK(dif_edn_set_auto_mode(&edn1, edn_params1));
+    EDN_TESTUTILS_WAIT_FOR_STATUS(&edn1, kDifEdnSmStateAutoAckWait, true,
+                                  kEdnBootModeTimeout);
+    // Re-enable ENTROPY_SRC in Non-FIPS mode.
+    CHECK_DIF_OK(dif_entropy_src_stop(&entropy_src));
+    entropy_src_config.fips_enable = false;
+    CHECK_DIF_OK(dif_entropy_src_configure(&entropy_src, entropy_src_config,
+                                           kDifToggleEnabled));
+    // Enable EDN0 in boot-time request mode.
+    CHECK_DIF_OK(dif_edn_set_boot_mode(&edn0));
+    CHECK_DIF_OK(dif_edn_configure(&edn0));
+    EDN_TESTUTILS_WAIT_FOR_STATUS(&edn0, kDifEdnSmStateBootGenAckWait, true,
+                                  kEdnBootModeTimeout);
+  }
+
+  if (round == 3) {
+    // Enable both EDNs in auto request mode.
+    CHECK_DIF_OK(dif_edn_set_auto_mode(&edn0, edn_params0));
+    CHECK_DIF_OK(dif_edn_set_auto_mode(&edn1, edn_params1));
+  }
+
+  return OK_STATUS();
+}
+
+static void consume_entropy(unsigned int round,
+                            dif_otbn_err_bits_t otbn_err_val,
+                            dif_rv_core_ibex_rnd_status_t ibex_rnd_fips) {
+  uint32_t ibex_rnd_data;
+  dif_rv_core_ibex_rnd_status_t ibex_rnd_status;
+  dif_otbn_irq_state_snapshot_t intr_state;
+  CHECK_STATUS_OK(entropy_config(round));
+  // Launch an OTBN program consuming entropy via both
+  // the RND and the URND interface.
+  CHECK_STATUS_OK(otbn_testutils_execute(&otbn));
+  // Verify that the OTBN finishes with the expected error values
+  // and interrupt flags.
+  CHECK_STATUS_OK(otbn_testutils_wait_for_done(&otbn, otbn_err_val));
+  CHECK_DIF_OK(dif_otbn_irq_get_state(&otbn, &intr_state));
+  CHECK(intr_state & 0x1);
+  CHECK_DIF_OK(dif_otbn_irq_acknowledge_all(&otbn));
+  // Read rnd data through the IBEX and verify if the FIPS compliance
+  // status is as expected.
+  // The first read gets rid of leftover entropy from previous configurations
+  // of the entropy complex.
+  CHECK_DIF_OK(dif_rv_core_ibex_read_rnd_data(&rv_core_ibex, &ibex_rnd_data));
+  IBEX_SPIN_FOR(rv_core_ibex_testutils_is_rnd_data_valid(&rv_core_ibex),
+                kEdnBootModeTimeout);
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_get_rnd_status(&rv_core_ibex, &ibex_rnd_status));
+  // The second read now contains the entropy from the current configuration.
+  CHECK_DIF_OK(dif_rv_core_ibex_read_rnd_data(&rv_core_ibex, &ibex_rnd_data));
+  CHECK((ibex_rnd_status & kDifRvCoreIbexRndStatusFipsCompliant) ==
+        ibex_rnd_fips);
+}
+
+bool test_main(void) {
+  init_peripherals();
+  // Prepare the OTBN for execution.
+  configure_otbn();
+  // Run the Procedure and check if EDN1 produces FIPS non-compliant entropy.
+  consume_entropy(1, kDifOtbnErrBitsRndFipsChkFail,
```

Comment:
Nit.
```suggestion
  consume_entropy(/*round=*/1, kDifOtbnErrBitsRndFipsChkFail,
```

Reply:
Thanks, I'll update this before merging.